### PR TITLE
[Buttons] add traitCollectionDidChangeBlock API.

### LINKED ISSUE #7854: [Buttons] Internal issue: b/137202741
This was filed as an internal issue. If you are a Googler, please visit [b/137202741](http://b/137202741) for more details.

---
#### Internal data
- Associated internal bug: [b/137202741](http://b/137202741)

### PR DESCRIPTION
**Pull request by fluxminds**
Provides an API for clients to respond to `traitCollectionDidChange` without subclassing.

closes https://github.com/material-components/material-components-ios/issues/7854

### CODE REVIEW

File: components/Buttons/src/MDCButton.m [hunk 1]
```diff
@@ -314,6 +314,15 @@ - (CGSize)sizeThatFits:(CGSize)size {
   return superSize;
 }
 
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
+  [super traitCollectionDidChange:previousTraitCollection];
+
+  [self updateTitleFont];
```
Comment (romoore) [inline components/Buttons/src/MDCButton.m]: Please don't add this call to this PR. The PR should focus on adding the block support. If we need to adjust fonts after trait collection changes, please put that in a separate PR.

**Review by romoore — DISMISSED** [review]
Looks like an extra change snuck-in.

File: components/Buttons/src/MDCButton.m (hunk 1 above)
Comment (fluxminds) [reply components/Buttons/src/MDCButton.m]: yea. It's reasonable. Addressed in 6902a069263e11b222182e4e9c34461d7df68271

**Review by romoore — APPROVED** [review]
